snuba/reader.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import re
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Mapping, Optional, Sequence
+
+from dateutil.tz import tz
+
+
+if TYPE_CHECKING:
+    from mypy_extensions import TypedDict
+
+    Column = TypedDict("Column", {"name": str, "type": str})
+    Result = TypedDict(
+        "Result",
+        {
+            "meta": Sequence[Column],
+            "data": Sequence[Mapping[str, Any]],
+            "totals": Mapping[str, Any],
+        },
+        total=False,
```

Comment:
Which key can be missing ?

Reply:
The `totals` row is only included if explicitly asked for by the request. The other two are required. I can [split this up and make `meta` and `data` required and `totals` optional](https://mypy.readthedocs.io/en/latest/more_types.html#mixing-required-and-non-required-items) if you'd like. (In all honesty, I was just lazy and figured this code would be short-lived — see the comment above.)